Occasionally can't resolve names from CDSW containers

This causes PriviledgedActionException due to lack of kerberos connection.

```
cdsw@3o14mmdf0mfs61y0:~$ hdfs dfs -ls
2017-08-02 09:11:31,274 WARN  [main] security.UserGroupInformation (UserGroupInformation.java:doAs(1920)) - PriviledgedActionException as:user1@HADOOP (auth:KERBEROS) cause:javax.security.sasl.SaslException: GSS initiate failed [Caused by GSSException: No valid credentials provided (Mechanism level: kerberos)]
2017-08-02 09:11:32,855 WARN  [main] security.UserGroupInformation (UserGroupInformation.java:doAs(1920)) - PriviledgedActionException as:user1@HADOOP (auth:KERBEROS) cause:javax.security.sasl.SaslException: GSS initiate failed [Caused by GSSException: No valid credentials provided (Mechanism level: kerberos)]
2017-08-02 09:11:32,856 WARN  [main] security.UserGroupInformation (UserGroupInformation.java:hasSufficientTimeElapsed(1403)) - Not attempting to re-login since the last re-login was attempted less than 60 seconds before. Last Login=1501665091278
2017-08-02 09:11:36,106 WARN  [main] security.UserGroupInformation (UserGroupInformation.java:doAs(1920)) - PriviledgedActionException as:user1@HADOOP (auth:KERBEROS) cause:javax.security.sasl.SaslException: GSS initiate failed [Caused by GSSException: No valid credentials provided (Mechanism level: kerberos)]
2017-08-02 09:11:36,106 WARN  [main] security.UserGroupInformation (UserGroupInformation.java:hasSufficientTimeElapsed(1403)) - Not attempting to re-login since the last re-login was attempted less than 60 seconds before. Last Login=1501665091278
2017-08-02 09:11:37,429 WARN  [main] security.UserGroupInformation (UserGroupInformation.java:doAs(1920)) - PriviledgedActionException as:user1@HADOOP (auth:KERBEROS) cause:javax.security.sasl.SaslException: GSS initiate failed [Caused by GSSException: No valid credentials provided (Mechanism level: kerberos)]
2017-08-02 09:11:37,429 WARN  [main] security.UserGroupInformation (UserGroupInformation.java:hasSufficientTimeElapsed(1403)) - Not attempting to re-login since the last re-login was attempted less than 60 seconds before. Last Login=1501665091278
2017-08-02 09:11:38,963 WARN  [main] security.UserGroupInformation (UserGroupInformation.java:doAs(1920)) - PriviledgedActionException as:user1@HADOOP (auth:KERBEROS) cause:javax.security.sasl.SaslException: GSS initiate failed [Caused by GSSException: No valid credentials provided (Mechanism level: kerberos)]
2017-08-02 09:11:38,963 WARN  [main] security.UserGroupInformation (UserGroupInformation.java:hasSufficientTimeElapsed(1403)) - Not attempting to re-login since the last re-login was attempted less than 60 seconds before. Last Login=1501665091278
2017-08-02 09:11:41,409 WARN  [main] security.UserGroupInformation (UserGroupInformation.java:doAs(1920)) - PriviledgedActionException as:user1@HADOOP (auth:KERBEROS) cause:javax.security.sasl.SaslException: GSS initiate failed [Caused by GSSException: No valid credentials provided (Mechanism level: kerberos)]
2017-08-02 09:11:41,410 WARN  [main] ipc.Client (Client.java:run(709)) - Couldn't setup connection for user1@HADOOP to ip-10-0-0-17.ap-northeast-1.compute.internal/10.0.0.17:8020
javax.security.sasl.SaslException: GSS initiate failed [Caused by GSSException: No valid credentials provided (Mechanism level: kerberos)]
        at com.sun.security.sasl.gsskerb.GssKrb5Client.evaluateChallenge(GssKrb5Client.java:211)
        at org.apache.hadoop.security.SaslRpcClient.saslConnect(SaslRpcClient.java:413)
        at org.apache.hadoop.ipc.Client$Connection.setupSaslConnection(Client.java:594)
        at org.apache.hadoop.ipc.Client$Connection.access$2000(Client.java:396)
        at org.apache.hadoop.ipc.Client$Connection$2.run(Client.java:761)
        at org.apache.hadoop.ipc.Client$Connection$2.run(Client.java:757)
        at java.security.AccessController.doPrivileged(Native Method)
        at javax.security.auth.Subject.doAs(Subject.java:422)
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1917)
        at org.apache.hadoop.ipc.Client$Connection.setupIOstreams(Client.java:756)
        at org.apache.hadoop.ipc.Client$Connection.access$3000(Client.java:396)
        at org.apache.hadoop.ipc.Client.getConnection(Client.java:1555)
        at org.apache.hadoop.ipc.Client.call(Client.java:1478)
        at org.apache.hadoop.ipc.Client.call(Client.java:1439)
        at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:230)
        at com.sun.proxy.$Proxy10.getFileInfo(Unknown Source)
        at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.getFileInfo(ClientNamenodeProtocolTranslatorPB.java:771)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:260)
        at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:104)
        at com.sun.proxy.$Proxy11.getFileInfo(Unknown Source)
        at org.apache.hadoop.hdfs.DFSClient.getFileInfo(DFSClient.java:2126)
        at org.apache.hadoop.hdfs.DistributedFileSystem$20.doCall(DistributedFileSystem.java:1262)
        at org.apache.hadoop.hdfs.DistributedFileSystem$20.doCall(DistributedFileSystem.java:1258)
        at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81)
        at org.apache.hadoop.hdfs.DistributedFileSystem.getFileStatus(DistributedFileSystem.java:1258)
        at org.apache.hadoop.fs.Globber.getFileStatus(Globber.java:64)
        at org.apache.hadoop.fs.Globber.doGlob(Globber.java:272)
        at org.apache.hadoop.fs.Globber.glob(Globber.java:151)
0 seconds before. Last Login=1501665091278
        at org.apache.hadoop.fs.FileSystem.globStatus(FileSystem.java:1715)
        at org.apache.hadoop.fs.shell.PathData.expandAsGlob(PathData.java:326)
        at org.apache.hadoop.fs.shell.Command.expandArgument(Command.java:235)
        at org.apache.hadoop.fs.shell.Command.expandArguments(Command.java:218)
        at org.apache.hadoop.fs.shell.FsCommand.processRawArguments(FsCommand.java:102)
        at org.apache.hadoop.fs.shell.Command.run(Command.java:165)
        at org.apache.hadoop.fs.FsShell.run(FsShell.java:315)
        at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:70)
        at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:84)
        at org.apache.hadoop.fs.FsShell.main(FsShell.java:372)
Caused by: GSSException: No valid credentials provided (Mechanism level: kerberos)
        at sun.security.jgss.krb5.Krb5Context.initSecContext(Krb5Context.java:775)
        at sun.security.jgss.GSSContextImpl.initSecContext(GSSContextImpl.java:248)
        at sun.security.jgss.GSSContextImpl.initSecContext(GSSContextImpl.java:179)
        at com.sun.security.sasl.gsskerb.GssKrb5Client.evaluateChallenge(GssKrb5Client.java:192)
        ... 41 more
Caused by: java.net.UnknownHostException: kerberos
        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:184)
        at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
        at java.net.Socket.connect(Socket.java:589)
        at sun.security.krb5.internal.TCPClient.<init>(NetClient.java:63)
        at sun.security.krb5.internal.NetClient.getInstance(NetClient.java:43)
        at sun.security.krb5.KdcComm$KdcCommunication.run(KdcComm.java:393)
        at sun.security.krb5.KdcComm$KdcCommunication.run(KdcComm.java:364)
        at java.security.AccessController.doPrivileged(Native Method)
        at sun.security.krb5.KdcComm.send(KdcComm.java:348)
        at sun.security.krb5.KdcComm.sendIfPossible(KdcComm.java:253)
        at sun.security.krb5.KdcComm.send(KdcComm.java:229)
        at sun.security.krb5.KdcComm.send(KdcComm.java:200)
        at sun.security.krb5.KrbTgsReq.send(KrbTgsReq.java:246)
        at sun.security.krb5.KrbTgsReq.sendAndGetCreds(KrbTgsReq.java:261)
        at sun.security.krb5.internal.CredentialsUtil.serviceCreds(CredentialsUtil.java:308)
        at sun.security.krb5.internal.CredentialsUtil.acquireServiceCreds(CredentialsUtil.java:126)
        at sun.security.krb5.Credentials.acquireServiceCreds(Credentials.java:458)
        at sun.security.jgss.krb5.Krb5Context.initSecContext(Krb5Context.java:693)
        ... 44 more
2017-08-02 09:11:41,415 WARN  [main] security.UserGroupInformation (UserGroupInformation.java:doAs(1920)) - PriviledgedActionException as:user1@HADOOP (auth:KERBEROS) cause:java.io.IOException: Couldn't s
etup connection for user1@HADOOP to ip-10-0-0-17.ap-northeast-1.compute.internal/10.0.0.17:8020
ls: Failed on local exception: java.io.IOException: Couldn't setup connection for user1@HADOOP to ip-10-0-0-17.ap-northeast-1.compute.internal/10.0.0.17:8020; Host Details : local host is: "3o14mmdf0mfs61
y0/100.66.0.6"; destination host is: "ip-10-0-0-17.ap-northeast-1.compute.internal":8020; 
```

--- a/cdsw-secure-cluster/instance-postcreate-cdsw.sh
+++ b/cdsw-secure-cluster/instance-postcreate-cdsw.sh
@@ -31,10 +31,12 @@ yum -y install git
 
 # install dnsmasq
 yum -y install dnsmasq
+cat /etc/resolv.conf | grep nameserver > /etc/dnsmasq.resolv.conf
+perl -pi -e "s|^.*?resolv-file.*?$|resolv-file=/etc/dnsmasq.resolv.conf|" /etc/dnsmasq.conf
 systemctl start dnsmasq
 
 # Add DNS(dnsmasq on local)
-perl -pi -e "s/nameserver/nameserver $(hostname -i)\nnameserver/" /etc/resolv.conf
+perl -pi -e "s/nameserver.*$/nameserver $(hostname -i)/" /etc/resolv.conf
 chattr +i /etc/resolv.conf 
 
 # This domain for DNS and is unrelated to Kerberos or LDAP domains.

